pkg/common/utils.go

```diff
@@ -479,12 +473,43 @@ func existingErrorCode(err error) (codes.Code, error) {
 	if err == nil {
 		return codes.Unknown, fmt.Errorf("null error")
 	}
-	if status, ok := status.FromError(err); ok {
-		return status.Code(), nil
+	var tmpError *TemporaryError
+	if errors.As(err, &tmpError) {
+		if status, ok := status.FromError(err); ok {
+			return status.Code(), nil
+		}
+	}
+	var googleErr *googleapi.Error
+	if !errors.As(err, &googleErr) {
+		if status, ok := status.FromError(err); ok {
+			return status.Code(), nil
+		}
 	}
```

Comment:
Why do we need `!errors.As(err, &googleErr)`?

Reply:
We want to catch any other errors that qualify as status-able. This is because there is yae (yet another error) that we can get back from compute engine:

https://github.com/grpc/grpc-go/blob/f7d3d3eecbeecfdc61451426545dea92981666eb/internal/status/status.go#L174

There's no direct way to access this type because its under internal, so i thought this would be the best next thing.